server/server.cpp

```diff
@@ -5,7 +5,51 @@
 #include <stdlib.h>
 #include <string>
 #include <sys/socket.h>
+#include <thread>
 #include <unistd.h>
+#include <vector>
+
+void handle_client(int client_socket, sockaddr_in remote_address) {
+  std::string client_ip = inet_ntoa(remote_address.sin_addr);
+  int remote_port = ntohs(remote_address.sin_port);
+
+  std::cout << "Accepted new client @ " << client_ip << ":" << remote_port
+            << std::endl;
+
+  int BUFFLEN = 1024;
+  char buffer[BUFFLEN];
+
+  while (1) {
+    memset(buffer, 0, BUFFLEN);
+    // 5 recv
+    int bytes_received = recv(client_socket, buffer, BUFFLEN - 1, 0);
```

Comment:
That is blocking call, server waits here till client sends some data
Use it together with select 
while (!exitConnection) {
       fd_set readfds;
       FD_ZERO(&readfds);
       FD_SET(client_socket, &readfds);

       struct timeval timeout;
       timeout.tv_sec = 1;  // 1 second timeout
       timeout.tv_usec = 0;

       int activity = select(client_socket + 1, &readfds, NULL, NULL, &timeout);

       if (activity < 0 && errno != EINTR) {
           // Handle error
           break;
       }

       if (activity == 0) {
           // Timeout occurred, loop to check exitConnection
           continue;
       }

       if (FD_ISSET(client_socket, &readfds)) {
           int bytes_received = recv(client_socket, buffer, BUFFLEN - 1, 0);
           if (bytes_received <= 0) {
               // Handle connection closed or error
               break;
           }
           // Process the received data
       }
   }